pkg/reconciler/reconciler.go

```diff
@@ -360,23 +360,27 @@ func WithUninstallAnnotations(as ...annotation.Uninstall) Option {
 	}
 }
 
-// WithPreHook is an Option that configures the reconciler to run the given
-// PreHook just before performing any actions (e.g. install, upgrade, uninstall,
-// or reconciliation).
-func WithPreHook(h hook.PreHook) Option {
+// WithExtension is an Option that registers an extension into the reconciler.
+// For example, extensions may be necessary when need arises to manage certain
+// resources outsides of the standard Helm-based workflow.
+func WithExtension(e extension.ReconcilerExtension) Option {
```

Comment:
The same suggestion as [here](https://github.com/operator-framework/helm-operator-plugins/pull/157#discussion_r826678288) for the design of extensions and `NoOpReconcilerExtension`